### PR TITLE
Default to `~/.config/gmailctl` for newly created config dirs

### PR DESCRIPTION
**Pull request by lgarron**
This is a minimal fix for https://github.com/mbrt/gmailctl/issues/144

### CODE REVIEW

File: cmd/gmailctl/cmd/root_cmd.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func init() {
 	// Here you will define your flags and configuration settings.
 	// Cobra supports persistent flags, which, if defined here,
 	// will be global for your application.
-	rootCmd.PersistentFlags().StringVar(&cfgDir, "config", "", "config directory (default is $HOME/.gmailctl)")
+	rootCmd.PersistentFlags().StringVar(&cfgDir, "config", "", "config directory (default is $HOME/.config/gmailctl unless $HOME/.gmailctl is already present)")
```
Comment (mbrt) [inline cmd/gmailctl/cmd/root_cmd.go:54]: The wording is 100% correct, but might be unnecessary detail to newcomers. I would perhaps word it as:

default search paths: $HOME/.gmailctl, $HOME/.config/gmailctl

**Review by mbrt — CHANGES_REQUESTED** [review]

File: cmd/gmailctl/cmd/root_cmd.go (hunk 1 above)
Comment (lgarron) [reply cmd/gmailctl/cmd/root_cmd.go:54]: Sounds good. Do you want to indicate that the latter will be created by default, or just leave implicit?

Comment (lgarron) [reply cmd/gmailctl/cmd/root_cmd.go:54]:
```suggestion
	rootCmd.PersistentFlags().StringVar(&cfgDir, "config", "", "config directory (default search paths: $HOME/.gmailctl, $HOME/.config/gmailctl)")
```

**Review by mbrt — CHANGES_REQUESTED** [review]
Thanks for the contributions!

There are also a bunch of references to the old directory in the main README.md. Could you update them?

I think the new behavior should also be explained in the `gmailctl init --help` [message](https://github.com/mbrt/gmailctl/blob/086c6a2bdbd8387f6b7cda94665501427abeee5c/cmd/gmailctl/cmd/init_cmd.go#L24-L26). Something along the lines that the default config directory location moved from one location to the other.

Sorry for the back and forth, I missed these points in the earlier review. Thanks!

**Comment by mbrt** [comment]
I thought about this a bit more and the problem with this PR is that it doesn't solve the problem for all platforms. I think the original issue is more about OSX and Windows, where polluting the home directory with dotfiles is not common.

I'm afraid this just introduces one more variant without really tackling the underlying issue.

**Comment by lgarron** [comment]
> I thought about this a bit more and the problem with this PR is that it doesn't solve the problem for all platforms. I think the original issue is more about OSX and Windows, where polluting the home directory with dotfiles is not common.
> 
> I'm afraid this just introduces one more variant without really tackling the underlying issue.

I mean, that's fine if you'd prefer to match OS conventions more closely. How would you feel about using https://github.com/adrg/xdg to compute the config path?

**Comment by mbrt** [comment]
Using `adrg/xdg` seems better to me. We would make the behavior more complicated, but at the same time behave in a more standard way. I also took a look at the library and it seems well maintained and low dependency.

This should be explained decently in the docs, but I would agree with that change.

**Comment by lgarron** [comment]
> Using `adrg/xdg` seems better to me. We would make the behavior more complicated, but at the same time behave in a more standard way. I also took a look at the library and it seems well maintained and low dependency.
> 
> This should be explained decently in the docs, but I would agree with that change.

It's been two years, but I finally got around to this: https://github.com/mbrt/gmailctl/pull/455

(I've been using a local build with this PR in the meanwhile.)

**Pull request closed without merging**